meta/SearchCloud.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace dokuwiki\plugin\struct\meta;
+
+
+/**
+ * Class SearchCloud
+ *
+ * The same as @see SearchConfig, but executed a search that is not pid-focused
+ *
+ * @package dokuwiki\plugin\struct\meta
+ */
+class SearchCloud extends SearchConfig {
+
+
+    /**
+     * Transform the set search parameters into a statement
+     *
+     * @return array ($sql, $opts) The SQL and parameters to execute
+     */
+    public function getSQL() {
+        if(!$this->columns) throw new StructException('nocolname');
+
+        $QB = new QueryBuilder();
+        reset($this->schemas);
+        $schema = current($this->schemas);
+        $datatable = 'data_' . $schema->getTable();
+        if(!$schema->isLookup()) {
+            $QB->addTable('schema_assignments');
+            $QB->filters()->whereAnd("$datatable.pid = schema_assignments.pid");
+            $QB->filters()->whereAnd("schema_assignments.tbl = '{$schema->getTable()}'");
+            $QB->filters()->whereAnd("schema_assignments.assigned = 1");
+            $QB->filters()->whereAnd("GETACCESSLEVEL($datatable.pid) > 0");
+            $QB->filters()->whereAnd("PAGEEXISTS($datatable.pid) = 1");
+        }
+        $QB->addTable($datatable);
+        $QB->filters()->whereAnd("$datatable.latest = 1");
+
+        $col = $this->columns[0];
+        if($col->isMulti()) {
+            $multitable = "multi_{$col->getTable()}";
+            $MN = 'M' . $col->getColref();
+
+            $QB->addLeftJoin(
+                $datatable,
+                $multitable,
+                $MN,
+                "$datatable.pid = $MN.pid AND
+                     $datatable.rev = $MN.rev AND
+                     $MN.colref = {$col->getColref()}"
+            );
+
+            $col->getType()->select($QB, $MN, 'value', 'tag');
+            $colname = $MN . '.value';
+        } else {
+            $col->getType()->select($QB, $datatable, $col->getColName(), 'tag');
+            $colname = $datatable . '.' . $col->getColName();
+        }
+        $QB->addSelectStatement("COUNT($colname)", 'count');
+        $QB->addGroupByStatement('tag');
+        $QB->addOrderBy('count DESC');
+
+        return $QB->getSQL();
+    }
+
+    /**
+     * Execute this search and return the result
+     *
+     * The result is a two dimensional array of Value()s.
+     *
+     * This will always query for the full result (not using offset and limit) and then
+     * return the wanted range, setting the count (@see getCount) to the whole result number
```

Comment:
True, but we also would need to add a `addLimit()` method to `meta\QueryBuilder` which may create some ambiguity over which approach to take.